simulation.cpp

```diff
@@ -397,30 +397,32 @@ void test_simulation() noexcept//!OCLINT test may be many
         assert(are_same_env_functions(identity_env.env_function_A, identity));
 
         auto identity_net = net_param{{1,1}, linear};
-        assert(net_behaves_like_the_function(identity_net, identity));
+        assert(net_behaves_like_the_function(change_all_weights(identity_net, 1), identity));
 
         auto identity_ind = ind_param{identity_net};
 
         int pop_size = 2;
         auto minimal_pop = pop_param{pop_size, 0, 0};
 
+        auto sim_p = sim_param{};
+        sim_p.selection_strength = 2;
+
 
         simulation s{all_params{
                 identity_env,
                 identity_ind,
                 minimal_pop,
-                sim_param{}
+                sim_p
             }};
 
         //change target value to match output of ind 0 net
         size_t best_ind = 0;
-        auto best_net = get_nth_ind_net(s, best_ind);
-        auto resp_best = response(get_nth_ind(s, best_ind))[0];
+        auto best_net = change_all_weights(get_nth_ind_net(s, best_ind), 1);
```

Comment:
I do. Because what you initialize in the beginning is a network parameter object, not a network itself. The network parameters do not actually have weights. 

The `net_behaves_like_the_function(change_all_weights(identity_net, 1), identity)`  I had above was a weird workaround where it figured out by itself it had to (temporarily) construct a network for the params and then change its weigths only for the assert, but that did not actually change anything to the networks inside the simulation constructed later. 
In the new version I have removed the short-workaround version so it should be more clear that the network used for that assert is just a temporary one that doesn't affect the simulation.

Reply:
But why don't you just assign `identity_net` to  best_net, and instead do the thing all over again? Moreover you do not make sure in this case that it behaves as intended.
You just need to revert the order of operation, which makes the test also more readable.
It should something like this:
```suggestion
        change_nth_ind_net(s, best_ind, identity_net);
        auto best_net = get_nth_ind_net(s, best_ind);
```